### PR TITLE
mds: update META_POP_READDIR/FETCH/STORE and cache_hit_rate

### PR DESCRIPTION
**Pull request by yongseokoh**
I've found that decay counters related to META_POP_READDIR/FETCH/STORE and cache_hit_rate reported by 'dump loads' are not updated. If these metrics are refreshed, administrators and MDBalancer can appropriately balance (or pin) subdirectories among MDS ranks.

Fixes: https://tracker.ceph.com/issues/51600
Signed-off-by: Yongseok Oh <yongseok.oh@linecorp.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by vshankar — APPROVED** [review]
LGTM

File: src/mds/CDir.cc [hunk 1]
```diff
@@ -1577,6 +1578,8 @@ void CDir::fetch(MDSContext *c, std::string_view want_dn, bool ignore_authpinnab
 
   if (mdcache->mds->logger) mdcache->mds->logger->inc(l_mds_dir_fetch);
 
+  mdcache->mds->balancer->hit_dir(this, META_POP_FETCH);
```
Comment (batrick) [inline src/mds/CDir.cc:1581]: How odd these were never wired up!! Great catch!

**Review by batrick — APPROVED** [review]
:100:

File: src/mds/CDir.cc (hunk 1 above)
Comment (yongseokoh) [reply src/mds/CDir.cc:1581]: Yep. They were defined, but they were not considered so far.

**Comment by batrick** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2021-July-16

**Pull request merged**